ouroboros-consensus-test/src/Test/Ouroboros/Consensus/ChainGenerator/Some.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+
+-- | Utilities that make it easier to declare types that merely existential
+-- quantify some type arguments, and actually for any other type that is not
+-- amenable to @deriving@
```

Comment:
TODO explain that Hackage's `some` package doesn't work for us here since our data types are quantifying multiple variables at a time and we reject the encoding overheading that would be necessary to use `some` in that context

Reply:
Done.